app/scripts/metamask-controller.js

```diff
@@ -1628,9 +1660,17 @@ export default class MetamaskController extends EventEmitter {
     this.preferencesController.setAddresses(newAccounts);
     newAccounts.forEach((address) => {
       if (!oldAccounts.includes(address)) {
-        const label = `${deviceName[0].toUpperCase()}${deviceName.slice(1)} ${
-          parseInt(index, 10) + 1
-        } ${hdPathDescription || ''}`.trim();
+        let label;
+        if (deviceName === DEVICE_NAMES.QR) {
+          const keyringName = keyring.getName();
+          label = `${keyringName[0].toUpperCase()}${keyringName.slice(1)} ${
+            parseInt(index, 10) + 1
+          } ${hdPathDescription || ''}`.trim();
+        } else {
+          label = `${deviceName[0].toUpperCase()}${deviceName.slice(1)} ${
+            parseInt(index, 10) + 1
+          } ${hdPathDescription || ''}`.trim();
+        }
```

Comment:
Since there's a lot of repeated code between the two, maybe we can create a utility function in this file, passing in `keyringName` or `deviceName`, since that appears to be the only difference.